File: QRDialogView/src/main/java/daniel/perez/qrdialogview/CalendarDialog.java
package daniel.perez.qrdialogview;

import android.app.DatePickerDialog;
import android.app.TimePickerDialog;
import android.content.Context;
import android.widget.TextView;

import java.util.Calendar;

import daniel.perez.core.Utils;

public class CalendarDialog {
    private static final String TAG = "CalendarDialog";
    private Context context;
    private int mMinute, mHour, mDay, mMonth, mYear;
    private TextView newTimeDateString;

    public CalendarDialog(Context context, TextView textView) {
        this.context = context;
        this.newTimeDateString = textView;
    }

    public void timeDialog() {
        final Calendar calendar = Calendar.getInstance();

        mHour = calendar.get(Calendar.HOUR_OF_DAY);
        mMinute = calendar.get(Calendar.MINUTE);

        TimePickerDialog timePickerDialog = new TimePickerDialog(context, (timePicker, hour, minute) -> {
            mHour = hour;
            mMinute = minute;
            newTimeDateString.setText(mHour + ":" + mMinute);
        }, mHour, mMinute, false);

        timePickerDialog.show();
    }

    public void dateDialog() {
        final Calendar calendar = Calendar.getInstance();
        mYear = calendar.get(Calendar.YEAR);
        mMonth = calendar.get(Calendar.MONTH);
        mDay = calendar.get(Calendar.DAY_OF_MONTH);

        DatePickerDialog datePickerDialog = new DatePickerDialog(context, (datePicker, year, month, day) -> {
            mYear = year;
            mMonth = month + 1;
            mDay = day;
            newTimeDateString.setText(mMonth + "-" + mDay + "-" + mYear);
        }, mYear, mMonth, mDay);
        datePickerDialog.show();
    }
}

File: QRDialogView/src/main/java/daniel/perez/qrdialogview/CalendarDialog.kt
package daniel.perez.qrdialogview

import android.content.Context
import daniel.perez.core.model.DateSetEvent
import daniel.perez.core.model.TimeSetEvent
import daniel.perez.qrdialogview.rx.ObservableDatePickerDialog
import daniel.perez.qrdialogview.rx.ObservableTimePickerDialog
import daniel.perez.qrdialogview.rx.dateSetEvents
import daniel.perez.qrdialogview.rx.timeSetEvents
import io.reactivex.rxjava3.core.Observable
import java.util.*

class CalendarDialog(private val context: Context)
{
    fun timeDialog(): Observable<TimeSetEvent>
    {
        val calendar = Calendar.getInstance()

        val observableDialog: ObservableTimePickerDialog = timeSetEvents(context, calendar[Calendar.HOUR], calendar[Calendar.MINUTE])
        observableDialog.show()
        return observableDialog.timePickerEvents()
    }

    fun dateDialog(): Observable<DateSetEvent>
    {1
        val calendar = Calendar.getInstance()
        val observableDatePickerDialog: ObservableDatePickerDialog = dateSetEvents(context, calendar[Calendar.YEAR], calendar[Calendar.MONTH], calendar[Calendar.DAY_OF_MONTH])
        observableDatePickerDialog.show()

        return observableDatePickerDialog.datePickerEvents()
    }

    companion object
    {
        private const val TAG = "CalendarDialog"
    }

}